src/script.rs

```diff
@@ -398,9 +400,12 @@ impl Script {
     }
 
     /// Whether a script can be proven to have no satisfying input
+    /// In elements, is_provably_unspendable is consensus critical
+    /// matches the implementation of CScript::IsUnspendable()
     pub fn is_provably_unspendable(&self) -> bool {
-        !self.0.is_empty() && (opcodes::All::from(self.0[0]).classify() == opcodes::Class::ReturnOp ||
-                               opcodes::All::from(self.0[0]).classify() == opcodes::Class::IllegalOp)
+        !self.0.is_empty() && opcodes::All::from(self.0[0]).classify() == opcodes::Class::ReturnOp
```

Comment:
This isn't quite right, the `CScript::IsUnspendable` check is for exactly `OP_RETURN` rather than anything classified as a returnop.

Reply:
Yup, this is incorrect. Good catch!